### PR TITLE
Add example for request chaining in body payload

### PR DESCRIPTION
**Pull request by sapiderman**
# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Implements #757 
2. Missing example for request chaining in body

## How did you implement / how did you fix it  
1.  Adding specific example where previous request is used in body
2. Test the example

## How to test  
1. Use the following monika.yaml
```yaml
probes:
  - id: probe-01
    name: 'body from resonse'
    interval: 10

    requests:
      - url: https://reqres.in/api/users/1
        method: GET
        timeout: 5000
        saveBody: false
        headers:
          Content-Type: application/json; charset=utf-8
      - url: https://reqres.in/api/login
        method: POST
        timeout: 1000
        headers:
          Content-Type: application/json; charset=utf-8
        body:
          email: '{{ responses.[0].body.data.email }}'
          password: password
```
2. Then run `monika -c monica.yaml -s 0`

### CODE REVIEW

**Review by lukman7788 — APPROVED** [review]

**Review by dennypradipta — APPROVED** [review]

File: docs/src/pages/guides/examples.md [hunk 1]
```diff
@@ -214,3 +214,56 @@ probes:
...
+      'email': 'george.bluth@reqres.in',
+      'first_name': 'George',
+      'last_name': 'Bluth',
+      'avatar': 'https://reqres.in/img/faces/1-image.jpg',
+    },
+  ....,
+}
+```
+
+Then your chain request to get the user body.email would look something like:
```
Comment (nicnocquee) [inline docs/src/pages/guides/examples.md:236]: Then you can use the user's email in the login request's `body` as follows

File: docs/src/pages/guides/examples.md [hunk 2]
```diff
@@ -214,3 +214,56 @@ probes:
...
+
+        alerts:
+          - query: response.status != 200
+            message: Http Response status code is not 200!
+notifications:
+  - id: unique-id-desktop
+    type: desktop
+```
+
+Note: Please do not forget the tick marks before and after the opening and closing double braces to explicitly indicate a string value. YAML parsers will generate warnings without it.
```
Comment (nicnocquee) [inline docs/src/pages/guides/examples.md:269]: i think it's better to call it `single quote` instead of tick mark

**Review by nicnocquee — CHANGES_REQUESTED** [review]

**Review by ilmiawan — APPROVED** [review]

File: docs/src/pages/guides/examples.md [hunk 3]
```diff
@@ -214,3 +214,56 @@ probes:
 ```
 
 Using the above configuration, Monika will perform a login request in the first request, then use the returned token in the Authorization header of the second request.
+
+### Pass Response Data to Request Body
+
+Continuing with the examples from wwwh.reqres.in above, say we would like to use the previous GET request to perform a POST /login. If the initial data from reqres returned is something like below:
```
[the remark this answers is not in the document]
Comment (sapiderman) [reply docs/src/pages/guides/examples.md]: oops typo. Fixed, thanks!

File: docs/src/pages/guides/examples.md (hunk 2 above)
Comment (sapiderman) [reply docs/src/pages/guides/examples.md]: fixed!

File: docs/src/pages/guides/examples.md (hunk 1 above)
Comment (sapiderman) [reply docs/src/pages/guides/examples.md]: fixed, thanks!

File: docs/src/pages/guides/examples.md [hunk 4]
```diff
@@ -214,3 +214,56 @@ probes:
 ```
 
 Using the above configuration, Monika will perform a login request in the first request, then use the returned token in the Authorization header of the second request.
+
+### Pass Response Data to Request Body
+
+Continuing with the examples from www.reqres.in above, say we would like to use the previous GET request to perform a POST /login. If the data from the initial request is something like below:
+
+```yaml
```
Comment (nicnocquee) [inline docs/src/pages/guides/examples.md:222]: i think this should be json

Comment (sapiderman) [reply docs/src/pages/guides/examples.md:222]: oh yaa correct!

**Pull request merged**